File: SphrLibrary/Entities/OpenmHealth/SchemaId.cs
using System.Runtime.Serialization;

namespace SphrLibrary.Entities.OpenmHealth
{
    /// <summary>
    /// Open mHealth SchemaId リソースを表します。
    /// </summary>
    [DataContract()]
    [Serializable()]
    public class SchemaId
    {
        [DataMember(Name = "namespace")]
        public string @namespace = string.Empty;
        [DataMember()]
        public string name = string.Empty;
        [DataMember()]
        public string version = string.Empty;

        public SchemaId() { }

        public SchemaId(string @namespace, string name, string version)
        {
            this.@namespace = @namespace;
            this.name = name;
            this.version = version;
        }
    }
}

File: References/Java/SphrLibrary-Java/lib/src/main/java/sphrlibrary/entities/openmhealth/SchemaId.java
package sphrlibrary.entities.openmhealth;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

/**
 * Open mHealth の SchemaId Entity
 */
@Data
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class SchemaId {

    /**
     * スキーマの名前空間
     */
    @NonNull
    private String namespace = "";

    /**
     * スキーマの名前
     */
    @NonNull
    private String name = "";

    /**
     * スキーマのバージョン
     */
    @NonNull
    private String version = "";
}
